### ISSUE TITLE
[Calendar] Screen reader is not announcing column headers while navigating in the calendar table

### ISSUE THREAD

**Issue by anowduru** [microsoft/fluentui#19371]
### Environment Information

- **Package version(s)**: V8
- **Browser and OS versions**: Internet Explorer (IE11), Microsoft Edge legacy, Windows 10

### Describe the issue:

![image](https://user-images.githubusercontent.com/18456231/129073404-b8226ca8-b039-4068-84a8-175072d3e8ba.png)

![image](https://user-images.githubusercontent.com/18456231/129073538-cdf35ee0-42e8-41ec-bded-8b9b87bad0cc.png)

### Please provide a reproduction of the issue in a codepen:

###Actual Result:
Narrator is simply announcing as "April 8, 2021, non-selected, read-only", With keyboard shortcuts (ctrl+shift+alt+up arrow) it is announcing as not in table.

###Expected Result:
Screen reader should announce the Column headers along while navigating between columns. 

If an object is in a data table, the occupied rows and columns, and any headers associated with those rows or columns, shall be programmatically determinable

###User Impact 
Visually impaired users who rely on Screen reader will be unaware of the column headers in the calendar table.

### Documentation describing expected behavior

MAS Rule:
Row, Column, and Headers
